src/main/java/net/minecraftforge/common/crafting/ConditionalRecipe.java

```diff
@@ -91,6 +88,11 @@ public T read(ResourceLocation recipeId, JsonObject json)
                     return (T)RecipeManager.deserializeRecipe(recipeId, JSONUtils.getJsonObject(ele.getAsJsonObject(), "recipe"));
                 idx++;
             }
+            if (JSONUtils.hasField(json, "optional"))
+                if (!JSONUtils.isBoolean(json, "optional"))
+                    throw new JsonSyntaxException("Invalid contitional recipe, the 'optional' field value must be Boolean (true/false).");
+                else if (JSONUtils.isBoolean(json, "optional"))
```

Comment:
This seems either redundant or the wrong method, since it's the same as above.

Reply:
Dang it, manual merging error. This has to be `getBoolean()`.